cli/ops/tls.rs

```diff
@@ -55,23 +55,20 @@ pub fn op_dial_tls(
   _zero_copy: Option<PinnedBuf>,
 ) -> Result<JsonOp, ErrBox> {
   let args: DialTLSArgs = serde_json::from_value(args)?;
-
-  // TODO(ry) Using format! is suboptimal here. Better would be if
-  // state.check_net and resolve_addr() took hostname and port directly.
-  let address = format!("{}:{}", args.hostname, args.port);
   let cert_file = args.cert_file;
 
-  state.check_net(&address)?;
   if let Some(path) = cert_file.clone() {
     state.check_read(&path)?;
   }
 
-  let mut domain = args.hostname;
+  state.check_net(&args.hostname, args.port)?;
```

Comment:
I suggest you keep the same order of checks (ie. move this `check_state` above `check_read`)